jobs/cache_maintenance/src/cache_maintenance/clean_hf_datasets_cache.py

```diff
@@ -0,0 +1,50 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 The HuggingFace Authors.
+
+import glob
+import logging
+import os
+import shutil
+from datetime import datetime, timedelta
+from typing import Any
+
+from libcommon.storage import StrPath
+
+
+def clean_hf_datasets_cache(hf_datasets_cache: StrPath, expired_time_interval_seconds: int) -> None:
+    """
+    Delete temporary cache directories from job runners with datasets cache.
+    """
+    logging.info("delete hf datasets cache")
+    pattern = f"{hf_datasets_cache}/*"
```

Comment:
what if `hf_datasets_cache` happens to be `""` for some configuration reason? How could we avoid deleting everything below the root? Maybe at least ensure it's non-empty? Maybe some other idea to avoid deleting wrong files by error?

Reply:
added a sanity check that raises an error if len(hf_datasets_cache) < 10

Reply:
OK, thanks!